scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -286,16 +308,86 @@ public void createSimulationForPlan(final PlanId planId)
+      if (idActFromInitialPlan != null) {
+        final var actFromInitialPlan = initialPlan.getActivityById(idActFromInitialPlan);
+        //if act was present in initial plan
+        final var schedulerActIntoMerlinAct = new MerlinActivityInstance(activity.getType().getName(), activity.getStartTime(), activity.getArguments());
+        final var activityInstanceId = idsFromInitialPlan.get(activity.getId());
+        if(!schedulerActIntoMerlinAct.equals(actFromInitialPlan.get())) {
+          //update if it has changed
+          updateActivity(planId, schedulerActIntoMerlinAct, activityInstanceId);
+        }
+        ids.put(activity, activityInstanceId);
+      } else {
+        //act was not present in initial plan, create new activity
+        toAdd.add(activity);
+      }
+    }
+    final var actsFromNewPlan = plan.getActivitiesById();
+    for (final var idActInInitialPlan : idsFromInitialPlan.keySet()) {
+      if (!actsFromNewPlan.containsKey(idActInInitialPlan)) {
+        //activity has been removed by the scheduler, delete it
+        deleteActivity(idsFromInitialPlan.get(idActInInitialPlan));
+      }
+    }
+
+    //Create
+    ids.putAll(createActivities(planId, toAdd));
+
+    return ids;
   }
 
+  public void deleteActivity(final ActivityInstanceId id)
+  throws MerlinServiceException, IOException, NoSuchPlanException
+  {
+    final var request = "delete_activity_by_pk( id : %d ){ id }".formatted(id);
+    final var response = postRequest(request).orElseThrow(() -> new NoSuchPlanException(null));
+    try {
+      response.getJsonObject("data").getJsonObject("delete_activity_by_pk").getJsonNumber("id").longValueExact();
+    } catch (ClassCastException | ArithmeticException e) {
+      throw new NoSuchPlanException(null);
+    }
+  }
+
+  public void updateActivity(final PlanId planId, final MerlinActivityInstance activity, final ActivityInstanceId id)
+  throws MerlinServiceException, NoSuchPlanException, IOException
+  {
+    final var argFormat = "%s: %s ";
+    final var argumentsSb = new StringBuilder();
+    for (final var arg : activity.arguments().entrySet()) {
+      final var name = arg.getKey();
+      var value = getGraphQLValueString(arg.getValue());
+      argumentsSb.append(argFormat.formatted(name, value));
+    }
+    final var updateReq = """
+        mutation {
+          update_activity_by_pk( pk_columns : { id : %d, plan_id : %d }, _set: {start_offset : \"%s\", arguments : { %s }}) {
+           affected_rows
+          }
+        }
+        """.formatted(id, planId, activity.startTimestamp().toString(), argumentsSb.toString());
+
+    final var response = postRequest(updateReq).orElseThrow(() -> new NoSuchPlanException(planId));
+    try {
+      response.getJsonObject("data").getJsonObject("update_activity_by_pk").getJsonNumber("id").longValueExact();
+    } catch (ClassCastException | ArithmeticException e) {
+      throw new NoSuchPlanException(planId);
```

Comment:
See my above comment about the other case. In this case, there are more reasons why it may fail (plan is absent or activity is absent) so we pick one. Failure at the class cast or arithmetic level only means that something is wrong with the request. In other words, we assume the json unpacking is right on our side and they are proxy exception for the real issue.